appyx-interactions/android/src/main/kotlin/com/bumble/appyx/interactions/sample/TestDriveExperiment.kt

```diff
@@ -102,21 +96,30 @@ fun TestDriveExperiment() {
             }
 
             val output = model.output.collectAsState().value
-            val targetProps = output.currentTargetState.elementState.toProps()
-            Box(modifier = Modifier
-                .size(60.dp)
-                .offset(targetProps.offset.value.x, targetProps.offset.value.y)
-                .border(2.dp, targetProps.backgroundColor.value)
-            ) {
-                Text(
-                    modifier = Modifier.align(Alignment.Center),
-                    text = when (output) {
-                        is Keyframes -> output.currentIndex.toString()
-                        is Update -> "X"
-                    },
-                    fontSize = 24.sp,
-                    color = Color.White
-                )
+            val targetState: State<TestDriveModel.State<NavTarget>?> =
+                when (output) {
+                    is Keyframes -> output.currentSegmentTargetStateFlow
+                        .collectAsState(null)
```

Comment:
What's the reason for this null (or even the branching)?

Reply:
1. Null is because currentSegmentTargetStateFlow is flow, and we do not know the initial value of it
2. Branching is because for keyframes current target state is changes and we have to observe it. But for update its static